tests/test_tracing.py

```diff
@@ -507,9 +529,18 @@ def setUp(self):
         self.mock_is_lambda_context.return_value = True
         self.addCleanup(patcher.stop)
 
-    def test_set_correlation_ids(self):
+    @patch.object(tracer, 'current_span', autospec=True)
+    @patch.object(tracer, 'trace', autospec=True)
+    def test_set_correlation_ids(self, mock_trace, mock_current_span):
+        mock_span = MagicMock()
+        mock_span.trace_id = 123
+        mock_span.span_id = 456
+        mock_trace.return_value = mock_span
+        mock_current_span.return_value = mock_span
```

Comment:
Using mocks is confusing and unnecessary. If you're worried about the random number generator creating different ids each time, try seeding the generator. Something like https://docs.python.org/3/library/random.html#random.seed